### ISSUE TITLE
AL09 misses some redundant aliases

### ISSUE THREAD

**Issue by dr-1** [sqlfluff/sqlfluff#6918]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

File `test.sql`:

```sql
SELECT
    City AS "City",
    Citycode AS "CityCode"
FROM Foo
```

Command:

`sqlfluff lint --dialect=tsql --rules=AL09,RF06,CP02 test.sql`

No errors are found.

This is correct for `"CityCode"` where the alias changes the case, but not correct for `"City"` which repeats the column name with no case change and should therefore be flagged by AL09 as a redundant alias.

Note that setting the `case_sensitive` option of RF06 to `False` does not solve the problem. The output of the above command is then

```
== [test.sql] FAIL
L:   2 | P:  13 | RF06 | Unnecessary quoted identifier "City".
                       | [references.quoting]
L:   3 | P:  17 | RF06 | Unnecessary quoted identifier "CityCode".
                       | [references.quoting]
All Finished!
```

This is correct for `"City"` (and after removing its unnecessary quotes AL09 correctly flags it as a redundant alias), but not correct for `"CityCode"` which should be quoted according to AL09 to make the case change explicit.

I guess this is a conflict between the two rules.

### Expected Behaviour

See above

### Observed Behaviour

See above

### How to reproduce

See above

### Dialect

tsql

### Version

`sqlfluff, version 3.4.0`
`Python 3.11.2`

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Thanks @dr-1 for submitting the issue - it's a really interesting one.

There are two issues mentioned here:
1. With the default settings, how the `City AS "City"` should be treated? You mentioned that it should be treated as the same columns, and hence flagged as the alias is the same as the original, and therefore removed.

    If you look at the [RF06 Rule](https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-RF06) definition says, that T-SQL is a an UPPERCASE dialect, an therefore it will preserve the quoted identifier. Here is a relevant part:

    > Examples where quotes are not safe to remove: ...Identifiers where the quotes are necessary to preserve case (e.g. "Foo" or "foo"),   

    This means the column `City` will be converted to UPPERCASE `CITY` according to the UPPERCASE dialect, and  `CITY` ≠ `City` and it will be preserved. And I know it sounds strange for the first sight.

    So in that definition what you see IS the designated behavior IMHO.

2. With `case_sensitive = False` setting how `Citycode AS "CityCode"` should be treated? You mentioned that this alias should be treated as different and should NOT be removed. 

    I think you are right, this is a bug which should be corrected.

**Comment by peterbud** [comment]
I'd suggest to use `--noqa` or [rule exclusion](https://docs.sqlfluff.com/en/stable/configuration/ignoring_configuration.html) in such cases.
